### DISCUSSION TITLE
Ultimate mode doesn't disable the iGPU

### DISCUSSION THREAD

**Discussion by Faishun** [seerge/g-helper#3071] [Q&A] [answered]
In armoury crate it does, but here it does not, which is kind of an issue. Due to MUX switch errors with black screens that I get when unplugged and using a battery, I still get blue screens when using ultimate mode, because, as I checked in the task manager, it doesn't disable the iGPU completely.

Please fix, it's important in the scenario if I want to use an external monitor from battery. This is a common MUX switch issue for Asus TUF laptops, and I would really want this application to disable it completely. Otherwise installing armoury crate...

**Comment by seerge** [comment, marked as answer]
Hello, Ultimate mode is not supposed to disabled iGPU (no matter what application you use)

Ultimate mode routes internal display directly to dGPU (turning MUX off). That's the only thing it does.

**Reply by Faishun** [reply]
> Hello, Ultimate mode is not supposed to disabled iGPU (no matter what application you use)
> 
> Ultimate mode routes internal display directly to dGPU (turning MUX off). That's the only thing it does.

That's weird. The thing is, when I used armoury crate Ultimate mode, I most definitely could confirm that the iGPU was not usable, even gone from the task manager, therefore I didn't get the system shutdown error thing. Is there any way it could be a feature to disable iGPU somehow? Or is it not possible

**Reply by seerge** [reply]
I don't think that's the case :) How do you even check that iGPU is disabled ?

iGPU is part of your APU (CPU + iGPU). It's always available. Just when you use Ultimate mode it's not driving internal display, so it's not being used by system most likely.

G-Helper uses same toggle for Standard / Ultimate as Armoury. It's just an ON / OFF switch. And as I mentioned above it's not supposed to disable iGPU.

**Reply by Faishun** [reply]
> I don't think that's the case :) How do you even check that iGPU is disabled ?
> 
> iGPU is part of your APU (CPU + iGPU). It's always available. Just when you use Ultimate mode it's not driving internal display, so it's not being used by system most likely.
> 
> G-Helper uses same toggle for Standard / Ultimate as Armoury. It's just an ON / OFF switch. And as I mentioned above it's not supposed to disable iGPU.

You can check whether it is disabled in device manager. Now I enabled eco mode and it doesn't seem to be shutting down anymore. I suspect the MUX switch technology, but there are suggestions to set Nvidia Optimus to the high quality graphics processor (dGPU). All of it happens only when on battery. When plugged in, it's perfectly fine

**Reply by seerge** [reply]
I think you are mixing multiple things

- `Eco` mode just disables dGPU. This is **not** a MUX switch. When Nvidia dGPU is disabled there is no Optimus at all.
- `Standard` mode enables dGPU, and leaves internal display routed to Optimus /iGPU. Then Nvidia Optimus on software level controls what GPU to use for internal display (you can change that in Nvidia Control Panel)
- `Ultimate` mode is an proprietary "hardware" MUX toggle from Asus that routes internal screen directly to dGPU. There is also no Optimus here

When you unplug in `Ultimate` mode nothing happens. dGPU still runs internal screen. And iGPU doesn't disappear from device manager :) 

![image](https://github.com/user-attachments/assets/d4511ddd-f3f8-44e3-b1d2-16b2439b1caa)

**Reply by ojas-exe** [reply]
> I think you are mixing multiple things
> 
> * `Eco` mode just disables dGPU. This is **not** a MUX switch. When Nvidia dGPU is disabled there is no Optimus at all.
> * `Standard` mode enables dGPU, and leaves internal display routed to Optimus /iGPU. Then Nvidia Optimus on software level controls what GPU to use for internal display (you can change that in Nvidia Control Panel)
> * `Ultimate` mode is an proprietary "hardware" MUX toggle from Asus that routes internal screen directly to dGPU. There is also no Optimus here
> 
> When you unplug in `Ultimate` mode nothing happens. dGPU still runs internal screen. And iGPU doesn't disappear from device manager :)
> 
> ![image](https://private-user-images.githubusercontent.com/5920850/365261261-d4511ddd-f3f8-44e3-b1d2-16b2439b1caa.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODY3ODQ2OTcsIm5iZiI6MTc4Njc4NDM5NywicGF0aCI6Ii81OTIwODUwLzM2NTI2MTI2MS1kNDUxMWRkZC1mM2Y4LTQ0ZTMtYjFkMi0xNmIyNDM5YjFjYWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDgxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjA4MTVUMDg1OTU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWM2MDQxYmUyNmY1YzdmOGMzZDhlNGZhYTRkNjMzYzAzYjBlOTUwOTdiMTJhNDE0ZWEzZWExZjg0ZDk5OGMzYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmcmVzcG9uc2UtY29udGVudC10eXBlPWltYWdlJTJGcG5nIn0.wofThmjLR_KH6rGbSy_DINu3fotF7Ilc2bxb2MfqcLQ)

i used to switch gpu modes using g-helper and i remember that the i gpu disappears from task manager when on ultimate mode but today i noticed it wasnt disappearing , is this normal or is my laptop MUX hardware malfunctioning,  its asus tuf A16 FA607NUG - RL189Ws. please reply

**Comment by ojas-exe** [comment]
i used to switch gpu modes using g-helper and i remember that the i gpu disappears from task manager when on ultimate mode but today i noticed it wasnt disappearing , is this normal or is my laptop MUX hardware malfunctioning, its asus tuf A16 FA607NUG - RL189Ws. please reply

**Reply by seerge** [reply]
@ojas-exe answer is unchanged, Ultimate mode does not disable any iGPUs and it never did.
